### DISCUSSION TITLE
Smooth scrolling issue in VIM

### DISCUSSION THREAD

**Discussion by jmagaram** [zed-industries/zed#21982] [Feature Requests]
Whenever I do scrolling in VIM mode with ctrl+d and ctrl+u it jumps up and down instantly. I find this very disconcerting and disorienting. I've completely lost my context and feel like I've randomly jumped somewhere else is the file. Is there some way to smoothly scroll a bit up and down without this happening? Maybe I want that feature where the cursor stays in one place but want to go up and down a few lines instead of just one.

**Comment by jmagaram** [comment]
I've been researching this a bit. VS Code has this feature - "Editor: Smooth Scrolling". It makes a difference because when I was using VS Code, scrolling with Ctrl+D and Ctrl+U felt ok. When I switched to Zed something didn't feel right and I constantly felt lost in my code and felt the need to turn on the outline. Now I know why. Smooth scrolling would be great, but a physics based scrolling would be better, where the scrolling speeds up and then gently slows down once you arrive at the destination. I think this would show off a unique capability in Zed - the game-like rendering.

You can get this kind of scrolling in VIM. A quick search turned up https://github.com/yuttie/comfortable-motion.vim and https://github.com/karb94/neoscroll.nvim

**Comment by 0x2CA** [comment]
This belongs to the animation category, and this addition is very easy, what you said about physics is just a distinction of easing.

It's worth mentioning that currently, copying highlighting is just a simple display setting, which only delays and then clears, and cannot achieve a smooth animation effect.

```rust
      editor.highlight_background::<HighlightOnYank>(
            &ranges_to_highlight,
            |colors| colors.editor_document_highlight_read_background,
            cx,
        );
        cx.spawn(|this, mut cx| async move {
            cx.background_executor()
                .timer(Duration::from_millis(highlight_duration))
                .await;
            this.update(&mut cx, |editor, cx| {
                editor.clear_background_highlights::<HighlightOnYank>(cx)
            })
            .ok();
        })
        .detach();
```

Related to:
[WIP Animations #1904](https://github.com/zed-industries/zed/pull/1904)
[Smooth scrolling #4355](https://github.com/zed-industries/zed/issues/4355)
[Add an animation to the LSP checking indicator #9463](https://github.com/zed-industries/zed/pull/9463)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
